### ISSUE TITLE
Error: http: superfluous response.WriteHeader call from github.com/minio/minio/cmd/logger.(*ResponseWriter).WriteHeader (audit.go:115)

### ISSUE THREAD

**Issue by Win32Sector** [minio/minio#10587]
## Current Behavior

I see in minio logs these errors:

2020-09-28T22:51:13.363285397Z API: SYSTEM()
2020-09-28T22:51:13.363289197Z Time: 22:51:13 UTC 09/28/2020
2020-09-28T22:51:13.363292197Z DeploymentID: 61e3aea3-bd60-4f68-a827-5acc1ee40765
2020-09-28T22:51:13.363295197Z Error: http: superfluous response.WriteHeader call from github.com/minio/minio/cmd/logger.(*ResponseWriter).WriteHeader (audit.go:115)
2020-09-28T22:51:13.363300598Z        1: github.com/minio/minio@/cmd/server-main.go:468:cmd.serverMain.func2()

What can it means?

## Your Environment
* Version used (`minio --version`): minio/minio:RELEASE.2020-05-28T23-29-21Z
* Server setup and configuration: AKS, Kubernetes 1.18.6

**Comment by klauspost** [comment]
@Win32Sector Does this also reproduce on a more recent version, ie latest for instance?

**Comment by vadmeste** [comment]
@Win32Sector if this happens during listing object v1 call, then this is most likely fixed in recent versions

**Comment by Win32Sector** [comment]
I tried to update version to latest today, but I see this log:


```
2020-10-01T16:21:49.555461847Z dataUpdateTracker: deprecated data version, updating.
2020-10-01T16:22:04.260615326Z Endpoint:  http://10.240.8.15:9000  http://127.0.0.1:9000
2020-10-01T16:22:04.260635628Z 
2020-10-01T16:22:04.260638928Z Browser Access:
2020-10-01T16:22:04.260641628Z    http://10.240.8.15:9000  http://127.0.0.1:9000
2020-10-01T16:22:04.260644329Z 
2020-10-01T16:22:04.260646829Z Object API (Amazon S3 compatible):
2020-10-01T16:22:04.260650029Z    Go:         https://docs.min.io/docs/golang-client-quickstart-guide
2020-10-01T16:22:04.260655229Z    Java:       https://docs.min.io/docs/java-client-quickstart-guide
2020-10-01T16:22:04.260658130Z    Python:     https://docs.min.io/docs/python-client-quickstart-guide
2020-10-01T16:22:04.260660830Z    JavaScript: https://docs.min.io/docs/javascript-client-quickstart-guide
2020-10-01T16:22:04.260663430Z    .NET:       https://docs.min.io/docs/dotnet-client-quickstart-guide
2020-10-01T16:22:23.770270731Z 2020/10/01 16:22:23 http: superfluous response.WriteHeader call from github.com/minio/minio/cmd/logger.(*ResponseWriter).WriteHeader (audit.go:115)
2020-10-01T16:22:46.417970271Z 2020/10/01 16:22:46 http: superfluous response.WriteHeader call from github.com/minio/minio/cmd/logger.(*ResponseWriter).WriteHeader (audit.go:115)
2020-10-01T16:23:16.427789016Z 2020/10/01 16:23:16 http: superfluous response.WriteHeader call from github.com/minio/minio/cmd/logger.(*ResponseWriter).WriteHeader (audit.go:115)
2020-10-01T16:23:46.876260294Z 2020/10/01 16:23:46 http: superfluous response.WriteHeader call from github.com/minio/minio/cmd/logger.(*ResponseWriter).WriteHeader (audit.go:115)
2020-10-01T16:24:16.302373792Z 2020/10/01 16:24:16 http: superfluous response.WriteHeader call from github.com/minio/minio/cmd/logger.(*ResponseWriter).WriteHeader (audit.go:115)
2020-10-01T16:24:46.336392726Z 2020/10/01 16:24:46 http: superfluous response.WriteHeader call from github.com/minio/minio/cmd/logger.(*ResponseWriter).WriteHeader (audit.go:115)
2020-10-01T16:25:16.440121871Z 2020/10/01 16:25:16 http: superfluous response.WriteHeader call from github.com/minio/minio/cmd/logger.(*ResponseWriter).WriteHeader (audit.go:115)
2020-10-01T16:25:46.650765832Z 2020/10/01 16:25:46 http: superfluous response.WriteHeader call from github.com/minio/minio/cmd/logger.(*ResponseWriter).WriteHeader (audit.go:115)
2020-10-01T16:26:16.252177236Z 2020/10/01 16:26:16 http: superfluous response.WriteHeader call from github.com/minio/minio/cmd/logger.(*ResponseWriter).WriteHeader (audit.go:115)
2020-10-01T16:26:46.661953802Z 2020/10/01 16:26:46 http: superfluous response.WriteHeader call from github.com/minio/minio/cmd/logger.(*ResponseWriter).WriteHeader (audit.go:115)
2020-10-01T16:26:49.616300897Z 
2020-10-01T16:26:49.616335899Z API: SYSTEM()
2020-10-01T16:26:49.616339800Z Time: 16:26:49 UTC 10/01/2020
2020-10-01T16:26:49.616342900Z DeploymentID: 61e3aea3-bd60-4f68-a827-5acc1ee40765
2020-10-01T16:26:49.616345700Z Error: cache version deprecated (will autoupdate)
2020-10-01T16:26:49.616348600Z        3: cmd/data-usage-cache.go:452:cmd.(*dataUsageCache).load()
2020-10-01T16:26:49.616366402Z        2: cmd/fs-v1.go:244:cmd.(*FSObjects).CrawlAndGetDataUsage()
2020-10-01T16:26:49.616369102Z        1: cmd/data-crawler.go:108:cmd.runDataCrawler()
2020-10-01T16:27:16.917465659Z 2020/10/01 16:27:16 http: superfluous response.WriteHeader call from github.com/minio/minio/cmd/logger.(*ResponseWriter).WriteHeader (audit.go:115)
2020-10-01T16:27:52.567845264Z 2020/10/01 16:27:52 http: superfluous response.WriteHeader call from github.com/minio/minio/cmd/logger.(*ResponseWriter).WriteHeader (audit.go:115)
2020-10-01T16:28:17.035856534Z 2020/10/01 16:28:17 http: superfluous response.WriteHeader call from github.com/minio/minio/cmd/logger.(*ResponseWriter).WriteHeader (audit.go:115)
2020-10-01T16:28:46.970800076Z 2020/10/01 16:28:46 http: superfluous response.WriteHeader call from github.com/minio/minio/cmd/logger.(*ResponseWriter).WriteHeader (audit.go:115)
2020-10-01T16:29:16.989224445Z 2020/10/01 16:29:16 http: superfluous response.WriteHeader call from github.com/minio/minio/cmd/logger.(*ResponseWriter).WriteHeader (audit.go:115)
```
.\mc.exe admin info minio
```
●  dev-minio:9000
   Uptime: 5 minutes 
   Version: 2020-09-26T03:44:56Z
   Network: 1/1 OK 
```

18 GiB Used, 52,936 Buckets, 74,607 Objects

**Comment by harshavardhana** [comment]
> 18 GiB Used, 52,936 Buckets, 74,607 Objects

Are you uploading one bucket per object? - I agree that there is no limit to the number of buckets.

But it is superfluous to create so many buckets.

**Comment by Win32Sector** [comment]
No, our workflow - it is creating one bucket for one workflow. 
One workflow - it is one bucket and 1 to 10000 files in it.

As I understand, it is limits for buckets number? 
Or it is only warning?

**Comment by harshavardhana** [comment]
> As I understand, it is limits for buckets number?
> Or it is only warning?

I meant don't do it @Win32Sector

**Comment by vadmeste** [comment]
> 2020-10-01T16:22:23.770270731Z 2020/10/01 16:22:23 http: superfluous response.WriteHeader call from github.com/minio/minio/cmd/logger.(*ResponseWriter).WriteHeader (audit.go:115)

@Win32Sector what you are doing when you see that log line ? (uploading a file, download, ..)

**Comment by Win32Sector** [comment]
> > 2020-10-01T16:22:23.770270731Z 2020/10/01 16:22:23 http: superfluous response.WriteHeader call from github.com/minio/minio/cmd/logger.(*ResponseWriter).WriteHeader (audit.go:115)
> 
> @Win32Sector what you are doing when you see that log line ? (uploading a file, download, ..)

Sorry for waiting.

Application tries to upload files to bucket.

**Comment by vadmeste** [comment]
@Win32Sector what arguments did you pass to MinIO server? also are you using federation ?

**Comment by Win32Sector** [comment]
No, I use standalone version in this case. 
I want to understand what can mean this error.
My application said - I can't create blob

**Comment by Win32Sector** [comment]
And what about bucket number? 
Why huge bucket number - it is a bad idea?

**Comment by vadmeste** [comment]
> My application said - I can't create blob

Not sure, could you paste `mc admin info` ?

I also need your `mc admin trace`, you can ping me at https://slack.min.io as well

**Comment by Win32Sector** [comment]
In random moments, our application can't create files in minio, 

Oct 6, 2020 @ 12:26:48.565 | Default skill applying failed
Oct 6, 2020 @ 12:26:46.375 | HTTP "GET" "/api/heartbeat/extractionTraining" responded 500 in 0.2260 ms

It these moments, I see errors as this in minio logs:

2020/10/06 08:26:41 http: superfluous response.WriteHeader call from github.com/minio/minio/cmd/logger.(*ResponseWriter).WriteHeader (audit.go:115)

I don't understand, what can I trace with mc client?


```
.\mc.exe admin info minio
●  pdev-r-minio:9000
   Uptime: 5 minutes 
   Version: 2020-09-26T03:44:56Z
   Network: 1/1 OK 

36 GiB Used, 71,203 Buckets, 164,096 Objects
```

**Comment by vadmeste** [comment]
> Oct 6, 2020 @ 12:26:46.375 | HTTP "GET" "/api/heartbeat/extractionTraining" responded 500 in 0.2260 ms

GET is a download operation and not upload. `mc admin trace` can show better what's going on.

**Comment by Win32Sector** [comment]
Ok, I'm got it

https://gist.github.com/Win32Sector/b4b93e086a517051051e83f808fea90f
https://gist.githubusercontent.com/Win32Sector/b4b93e086a517051051e83f808fea90f/raw/782400dd68bdf1b4c2c2337b8f018a8ef23d73c8/mc-log

**Comment by Win32Sector** [comment]
I find when minio works and suddenldy 503 error is appeared
https://gist.github.com/Win32Sector/4b056b7a8dd2889c65062275b0241d35

**Comment by vadmeste** [comment]
> I find when minio works and suddenldy 503 error is appeared
> https://gist.github.com/Win32Sector/4b056b7a8dd2889c65062275b0241d35

is there still a correspondance between the timestamp of these traces and the one of the superfluous error line ?

**Comment by Win32Sector** [comment]
As I see before 503 errors, there are multiple superfluous error lines then, when I see 503 errors in logs, minio container restarting status.

**Comment by Win32Sector** [comment]
As I see before 503 errors, there are multiple superfluous error lines then, when I see 503 errors in logs, minio container restarting status.

**Comment by vadmeste** [comment]
> As I see before 503 errors, there are multiple superfluous error lines then, when I see 503 errors in logs, minio container restarting status.

Try to increase kubernetes healthcheck timeout to avoid container restart. Otherwise we need more leads to find the superfluous error, at least what http handler (s3.HeadObject, s3.PutObject, ..) is causing it.

**Comment by Win32Sector** [comment]
"kubernetes healthcheck" you are mean liveness probe for minio?

**Comment by vadmeste** [comment]
> "kubernetes healthcheck" you are mean liveness probe for minio?

oh yes correct

**Comment by Win32Sector** [comment]
Ok, I'll try today and write about the results this week.

**Comment by harshavardhana** [comment]
Fixed by #10642
